src/adu_types/src/adu_types.c

```diff
@@ -0,0 +1,45 @@
+/**
+ * @file adu_type.h
+ * @brief Defines common data types used throughout ADU project.
+ *
+ * @copyright Copyright (c) Microsoft Corporation.
+ */
+
+#include <aduc/adu_types.h>
+#include <stdlib.h>
+#include <string.h>
+
+void ADUC_ConnectionInfoDeAlloc(ADUC_ConnectionInfo* info)
+{
+    free(info->connectionString);
+    info->connectionString = NULL;
+
+    free(info->certificateString);
+    info->certificateString = NULL;
+
+    free(info->opensslEngine);
+    info->opensslEngine = NULL;
+
+    free(info->opensslPrivateKey);
+    info->opensslPrivateKey = NULL;
+
+    info->authType = ADUC_AuthType_NotSet;
+    info->connType = ADUC_ConnType_NotSet;
+}
+
+ADUC_ConnType GetConnTypeFromConnectionString(const char* connectionString)
+{
+    ADUC_ConnType result = ADUC_ConnType_NotSet;
+
+    if (strstr(connectionString, "DeviceId=") != NULL)
+    {
```

Comment:
relatively simple validation, will this work for all cases, e.g.

ModuleId="jeff;DeviceId=32"

or are we okay with basic level of parsing?

Reply:
oh, it looks like ModuleId is only valid if DeviceId is valid? See nox comment about moving inside block

---
In reply to: [591973835](https://github.com/Azure/iot-hub-device-update/pull/12#discussion_r591973835) [](ancestors = 591973835)